src/components/drs/pages/DrsShow.js

```diff
@@ -0,0 +1,231 @@
+import '@fontsource/roboto';
+import React, {
+  useState, 
+  useEffect
+} from 'react';
+import axios from 'axios';
+import { 
+  Typography, 
+  Container, 
+  Table,
+  TableContainer, 
+  TableHead, 
+  TableBody, 
+  TableRow, 
+  TableCell, 
+  List, 
+  ListItem, 
+  ListItemText
+} from '@material-ui/core';
+import {
+    useParams, 
+    Link
+} from "react-router-dom";
+
+const ChildProperties = (props) => {
+  const childPropertyKey = props.childPropertiesArray[0];
+  const childPropertyValueArray = props.childPropertiesArray[1];
+
+  if(childPropertyKey === 'drs_object_children' || childPropertyKey === 'drs_object_parents'){
+    const drsObject = childPropertyValueArray.map((drs) => 
+    {
+      return(
+        <ListItem key={drs.id}>
+          <ListItemText secondary={`Name: ${drs.name}`}>
+            <Link to={`/drs/${drs.id}`}>
+              <Typography>ID: {drs.id}</Typography>
+            </Link>
+          </ListItemText>
+        </ListItem>
+      )
+    })
+    return(
+      <List>{drsObject}</List>
+    )
+  }
+  else {
+    const arrayObjects = childPropertyValueArray.map((object) => 
+    {
+      if (typeof object === 'string') {
+        return (
+          <ListItem key={object}>
+            <ListItemText primary={object}></ListItemText>
+          </ListItem>
+        );
+      }
+      else {
+        let objectProperties = Object.entries(object);
+        const properties = objectProperties.map((property) => 
+        {
+          return(
+            <ListItem key={property[1]}>
+              <ListItemText primary={`${property[0]}: ${property[1]}`}></ListItemText>
+            </ListItem>
+          );
+        });
+        return (
+          <List key={objectProperties}>{properties}</List>
+        );
+      }
+    });
+    return(
+      <List>{arrayObjects}</List>
+    );
+  }
+}
+
+const DrsDetailsRows = (props) => {
```

Comment:
this rendering is a bit too general, and leads to confusion. we know the response payload is a `DrsObject` with expected properties and value types. Treating the response as just a `map` and iterating through the keys doesn't allow you to treat certain properties uniquely. Right now, all array or object properties of a `DrsObject` are collapsed into a single table cell, which could be formatted better if you rendered a checksum list (or any other list)  in a special manner

Reply:
I suggest creating a `DrsObject` component that can handle certain properties uniquely, ie "this is how the checksum list should be rendered," "this is how the child list is rendered" vs collapsing everything into a single table cell